aptos-move/block-executor/src/scheduler.rs

```diff
@@ -473,8 +526,11 @@ impl Scheduler {
     fn set_executed_status(&self, txn_idx: TxnIndex, incarnation: Incarnation) {
         let mut status = self.txn_status[txn_idx].lock();
 
-        // Only makes sense when the current status is 'Executing'.
-        debug_assert!(*status == TransactionStatus::Executing(incarnation));
+        if !self.done() {
```

Comment:
This invariant shouldn't change due to the way we do early termination, so it's actually better to keep as is. Also avoid extra atomic variable read for nothing.